bib/cmd/bootc-image-builder/image_test.go

```diff
@@ -448,3 +449,234 @@ func TestGenPartitionTableDiskCustomizationRunsValidateLayoutConstraints(t *test
+			5 * datasizes.GiB,
+			[]blueprint.PartitionCustomization{
+				{
+					Type:    "btrfs",
+					MinSize: 10 * datasizes.GiB,
+					BtrfsVolumeCustomization: blueprint.BtrfsVolumeCustomization{
+						Subvolumes: []blueprint.BtrfsSubvolumeCustomization{
+							{
+								Mountpoint: "/",
+								Name:       "rootvol",
+							},
+						},
+					},
+				},
+			},
+			10 * datasizes.GiB,
+		},
+		// lvm
+		{
+			"lvm, no root minsize, expands to rootfsMinSize",
+			5 * datasizes.GiB,
+			[]blueprint.PartitionCustomization{
+				{
+					Type:    "lvm",
+					MinSize: 10 * datasizes.GiB,
+					VGCustomization: blueprint.VGCustomization{
+						LogicalVolumes: []blueprint.LVCustomization{
+							{
+								MinSize: 10 * datasizes.GiB,
+								FilesystemTypedCustomization: blueprint.FilesystemTypedCustomization{
+									Mountpoint: "/var",
+									FSType:     "xfs",
+								},
+							},
+						},
+					},
+				},
+			},
+			5 * datasizes.GiB,
+		},
+		{
+			"lvm, small root minsize, expands to rootfsMnSize",
+			5 * datasizes.GiB,
+			[]blueprint.PartitionCustomization{
+				{
+					Type:    "lvm",
+					MinSize: 1 * datasizes.GiB,
+					VGCustomization: blueprint.VGCustomization{
+						LogicalVolumes: []blueprint.LVCustomization{
+							{
+								MinSize: 1 * datasizes.GiB,
+								FilesystemTypedCustomization: blueprint.FilesystemTypedCustomization{
+									Mountpoint: "/",
+									FSType:     "xfs",
+								},
+							},
+						},
+					},
+				},
+			},
+			5 * datasizes.GiB,
+		},
+		{
+			"lvm, big root minsize",
+			5 * datasizes.GiB,
+			[]blueprint.PartitionCustomization{
+				{
+					Type:    "lvm",
+					MinSize: 10 * datasizes.GiB,
+					VGCustomization: blueprint.VGCustomization{
+						LogicalVolumes: []blueprint.LVCustomization{
+							{
+								MinSize: 10 * datasizes.GiB,
+								FilesystemTypedCustomization: blueprint.FilesystemTypedCustomization{
+									Mountpoint: "/",
+									FSType:     "xfs",
+								},
+							},
+						},
+					},
+				},
+			},
+			10 * datasizes.GiB,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			cnf := &bib.ManifestConfig{
+				Architecture:  arch.FromString("amd64"),
+				RootFSType:    "xfs",
+				RootfsMinsize: tc.rootfsMinSize,
+			}
+			cus := &blueprint.Customizations{
+				Disk: &blueprint.DiskCustomization{
+					Partitions: tc.partitions,
+				},
+			}
+			pt, err := bib.GenPartitionTable(cnf, cus, rng)
+			assert.NoError(t, err)
+
+			var rootSize uint64
+			err = pt.ForEachMountable(func(mnt disk.Mountable, path []disk.Entity) error {
+				if mnt.GetMountpoint() == "/" {
+					for idx := len(path) - 1; idx >= 0; idx-- {
+						if parent, ok := path[idx].(disk.Sizeable); ok {
+							rootSize = parent.GetSize()
+							break
+						}
+					}
+				}
+				return nil
+			})
+			assert.NoError(t, err)
+			// expected size is within a reasonable limit
+			assert.True(t, rootSize >= tc.expectedMinRootSize, rootSize < tc.expectedMinRootSize+500*datasizes.MiB)
```

Comment:
This can probably a bit too much,we can tweak this in a followup, but:
```go
-                       assert.True(t, rootSize >= tc.expectedMinRootSize, rootSize < tc.expectedMinRootSize+500*datasizes.MiB)
+                       assert.True(t, rootSize >= tc.expectedMinRootSize, rootSize < tc.expectedMinRootSize+2*datasizes.MiB)
```
or we are precise and do:
```go
$ git diff
diff --git a/bib/cmd/bootc-image-builder/image_test.go b/bib/cmd/bootc-image-builder/image_test.go
index c4416fd..2a9d59c 100644
--- a/bib/cmd/bootc-image-builder/image_test.go
+++ b/bib/cmd/bootc-image-builder/image_test.go
@@ -477,7 +477,7 @@ func TestGenPartitionTableDiskCustomizationSizes(t *testing.T) {
                        "empty disk customizaton, root expands to rootfsMinsize",
                        2 * datasizes.GiB,
                        nil,
-                       2 * datasizes.GiB,
+                       2*datasizes.GiB + 1031680,
                },
                // plain
                {
@@ -492,7 +492,7 @@ func TestGenPartitionTableDiskCustomizationSizes(t *testing.T) {
                                        },
                                },
                        },
-                       5 * datasizes.GiB,
+                       5*datasizes.GiB + 1031680,
                },
                {
                        "plain, small root minsize, expands to rootfsMnSize",
@@ -506,7 +506,7 @@ func TestGenPartitionTableDiskCustomizationSizes(t *testing.T) {
                                        },
                                },
                        },
-                       5 * datasizes.GiB,
+                       5*datasizes.GiB + 1031680,
                },
                {
                        "plain, big root minsize",
@@ -520,7 +520,7 @@ func TestGenPartitionTableDiskCustomizationSizes(t *testing.T) {
                                        },
                                },
                        },
-                       10 * datasizes.GiB,
+                       10*datasizes.GiB + 1031680,
                },
                // btrfs
                {
@@ -675,8 +675,7 @@ func TestGenPartitionTableDiskCustomizationSizes(t *testing.T) {
                                return nil
                        })
                        assert.NoError(t, err)
-                       // expected size is within a reasonable limit
-                       assert.True(t, rootSize >= tc.expectedMinRootSize, rootSize < tc.expectedMinRootSize+500*datasizes.MiB)
+                       assert.Equal(t, rootSize, tc.expectedMinRootSize)
                })
        }
 }

```
but I don't know where this number comes from?